src/Uno.UI/UI/Xaml/Controls/ItemCollection.cs

```diff
@@ -158,30 +158,31 @@ internal void SetItemsSource(object itemsSource)
 			}
 			else
 			{
-				var unwrappedSource = UnwrapItemsSource(itemsSource);
-
-				if (unwrappedSource is IList itemsSourceList)
+				object listSource = null;
+				if (itemsSource is IList<object> itemsSourceGenericList)
 				{
-					_itemsSource = itemsSourceList;
+					listSource = itemsSourceGenericList;
+					_itemsSource = itemsSourceGenericList;
 				}
-				else if (unwrappedSource is IEnumerable itemsSourceEnumerable)
+				else if (itemsSource is IList itemsSourceList)
 				{
-					_itemsSource = itemsSourceEnumerable.ToObjectArray();
+					listSource = itemsSourceList;
+					_itemsSource = new UntypedListWrapper(itemsSourceList);
 				}
 				else
 				{
-					throw new InvalidOperationException("Only IList- or IEnumerable-based ItemsSource is supported.");
+					_itemsSource = itemsSource.ToObjectArray();
 				}
 
-				ObserveCollectionChanged();
+				ObserveCollectionChanged(listSource);
 			}
 
-			VectorChanged?.Invoke(this, new VectorChangedEventArgs(CollectionChange.Reset, 0));
+			VectorChanged?.Invoke(this, new VectorChangedEventArgs(CollectionChange.Reset, 0)); // TODO now: this may fuck things up per frere Baptiste
```

Comment:
```suggestion
			VectorChanged?.Invoke(this, new VectorChangedEventArgs(CollectionChange.Reset, 0)); // TODO now: this may mess things up per frere Baptiste
```

Reply:
:D this is what I get when I forget to do Ctrl-Shift-F